security-oauth2-quickstart/src/test/java/org/acme/security/oauth2/TokenSecuredResourceTest.java

```diff
@@ -0,0 +1,76 @@
+package org.acme.security.oauth2;
+
+import io.quarkus.test.junit.QuarkusTest;
+import io.restassured.RestAssured;
+import org.junit.ClassRule;
+import org.junit.jupiter.api.Test;
+import org.testcontainers.containers.BindMode;
+import org.testcontainers.containers.FixedHostPortGenericContainer;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+@Testcontainers
+@QuarkusTest
+public class TokenSecuredResourceTest {
+
+    private static final String KEYCLOAK_SERVER_URL = System.getProperty("keycloak.url", "http://localhost:8180/auth");
```

Comment:
So we cannot use TestContainer like this, as the lifecycle is different from the Quarkus lifecycle.

You need to use a Quarkus Test Resource, and BTW, it will also let you set some properties dynamically. 
Check how the Kafka quickstart is built as an example.

Reply:
I will check how the Kafka quickstart is built.